src/main/java/com/gooddata/GoodDataEndpoint.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (C) 2004-2016, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata;
+
+import org.apache.http.HttpHost;
+
+import static com.gooddata.util.Validate.notEmpty;
+
+/**
+ * GoodData Platform endpoint represented by host, port and protocol
+ */
+public class GoodDataEndpoint {
+
+    private static final String PROTOCOL = "https";
+    private static final int PORT = 443;
+
+    private final String hostname;
+    private final int port;
+    private final String protocol;
+
+    /**
+     * Create GoodData endpoint for given hostname, port and protocol
+     * @param hostname
+     * @param port
+     * @param protocol
+     */
+    public GoodDataEndpoint(String hostname, int port, String protocol) {
+        this.hostname = notEmpty(hostname, "hostname");
+        this.port = port;
+        this.protocol = notEmpty(protocol, "protocol");
+    }
+
+    /**
+     * Create GoodData endpoint for given hostname, port using HTTPS protocol
+     * @param hostname
+     * @param port
+     */
+    public GoodDataEndpoint(String hostname, int port) {
+        this(hostname, port, PROTOCOL);
+    }
+
+    /**
+     * Create GoodData endpoint for given hostname using 443 port and HTTPS protocol
+     * @param hostname
+     */
+    public GoodDataEndpoint(String hostname) {
+        this(hostname, PORT, PROTOCOL);
+    }
+
+    public HttpHost toHttpHost() {
+        return new HttpHost(hostname, port, protocol);
+    }
+
+    public String toUri() {
+        return toHttpHost().toURI();
+    }
+
+    public String getHostname() {
```

Comment:
it is good custom in java world be as much convenient to users of my code as possible. I believe, that at least `toHttpHost()` and `toUri()` deserve to be documented thoroughly